examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/ClientCacheExampleActivity.java

```diff
@@ -0,0 +1,150 @@
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.TimeUnit;
+
+public class ClientCacheExampleActivity extends AppCompatActivity {
+  private static final int CACHE_SIZE_IN_BYTES = 1 * 1024 * 1024; // 1MB
+  private Button mSendButton;
+  private EditText mHostEdit;
+  private EditText mPortEdit;
+  private EditText mMessageEdit;
+  private TextView mResultText;
+  private CheckBox getCheckBox;
+  private CheckBox noCacheCheckBox;
+  private CheckBox onlyIfCachedCheckBox;
+  private ManagedChannel mChannel;
+  private SafeMethodCachingInterceptor.Cache cache;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_clientcacheexample);
+    mSendButton = (Button) findViewById(R.id.send_button);
+    mHostEdit = (EditText) findViewById(R.id.host_edit_text);
+    mPortEdit = (EditText) findViewById(R.id.port_edit_text);
+    mMessageEdit = (EditText) findViewById(R.id.message_edit_text);
+    getCheckBox = (CheckBox) findViewById(R.id.get_checkbox);
+    noCacheCheckBox = (CheckBox) findViewById(R.id.no_cache_checkbox);
+    onlyIfCachedCheckBox = (CheckBox) findViewById(R.id.only_if_cached_checkbox);
+    mResultText = (TextView) findViewById(R.id.grpc_response_text);
+    mResultText.setMovementMethod(new ScrollingMovementMethod());
+
+    cache = SafeMethodCachingInterceptor.newLruCache(CACHE_SIZE_IN_BYTES);
+  }
+
+  public void sendMessage(View view) {
+    ((InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE))
+        .hideSoftInputFromWindow(mHostEdit.getWindowToken(), 0);
+    mSendButton.setEnabled(false);
+    new GrpcTask().execute();
+  }
+
+  private class GrpcTask extends AsyncTask<Void, Void, String> {
+    private String mHost;
+    private String mMessage;
+    private int mPort;
+    private ManagedChannel mChannel;
+
+    @Override
+    protected void onPreExecute() {
+      mHost = mHostEdit.getText().toString();
+      mMessage = mMessageEdit.getText().toString();
+      String portStr = mPortEdit.getText().toString();
+      mPort = TextUtils.isEmpty(portStr) ? 0 : Integer.valueOf(portStr);
+      mResultText.setText("");
+    }
+
+    @Override
+    protected String doInBackground(Void... nothing) {
+      try {
+        mChannel = ManagedChannelBuilder.forAddress(mHost, mPort).usePlaintext(true).build();
+        Channel channelToUse =
+            ClientInterceptors.intercept(
+                ManagedChannelBuilder.forAddress(mHost, mPort).usePlaintext(true).build(),
+                SafeMethodCachingInterceptor.newSafeMethodCachingInterceptor(cache));
+        HelloRequest message = HelloRequest.newBuilder().setName(mMessage).build();
+        HelloReply reply;
+        if (getCheckBox.isChecked()) {
+          MethodDescriptor<HelloRequest, HelloReply> safeCacheableUnaryCallMethod =
+              GreeterGrpc.METHOD_SAY_HELLO.toBuilder().setSafe(true).build();
```

Comment:
uppercase static field is deprecated now.

Reply:
Updated